sw/device/lib/crypto/include/key_transport.h

```diff
@@ -80,73 +80,76 @@ crypto_status_t otcrypto_hw_backed_key(uint32_t version, const uint32_t salt[7],
                                        crypto_blinded_key_t *key);
 
 /**
- * Builds an unblinded key struct from a user (plain) key.
+ * Imports a user provided key to an unblinded key struct.
+ *
+ * This API takes as input a plain key from the user and writes it to the
+ * `unblinded_key.key`.
+ *
+ * The caller should populate the `unblinded_key.key_mode` and allocate
+ * space for the key. The value in the `checksum` field of the unblinded
+ * key struct will be populated by the key generation function.
  *
  * @param plain_key Pointer to the user defined plain key.
- * @param key_mode Crypto mode for which the key usage is intended.
  * @param[out] unblinded_key Generated unblinded key struct.
- * @return Result of the build unblinded key operation.
+ * @return Result of the unblinded key import operation.
  */
-crypto_status_t otcrypto_build_unblinded_key(
-    crypto_const_byte_buf_t plain_key, key_mode_t key_mode,
-    crypto_unblinded_key_t unblinded_key);
+crypto_status_t otcrypto_import_unblinded_key(
+    const crypto_const_word32_buf_t plain_key,
+    crypto_unblinded_key_t *unblinded_key);
 
 /**
- * Builds a blinded key struct from a plain key.
+ * Imports a user provided masked key in shares, to a blinded key struct.
  *
- * This API takes as input a plain key from the user and masks it using an
- * implantation specific masking with ‘n’ shares writing the output to
- * `blinded_key.keyblob`.
+ * This API takes as input a masked key from the user in two shares, and masks
+ * it using an implantation specific masking with ‘n’ shares writing the output
+ * to the `blinded_key.keyblob`.
  *
  * The caller should allocate and partially populate the blinded key struct,
  * including populating the key configuration and allocating space for the
- * keyblob. The caller should indicate the length of the allocated keyblob;
- * this function will return an error if the keyblob length does not match
- * expectations. For hardware-backed keys, the keyblob length is 0 and the
- * keyblob pointer may be `NULL`. For non-hardware-backed keys, the keyblob
- * should be twice the length of the key. The value in the `checksum` field of
- * the blinded key struct will be populated by the key generation function.
- *
- * @param plain_key Pointer to the user defined plain key.
- * @param[out] blinded_key Destination blinded key struct.
- * @return Result of the build blinded key operation.
+ * keyblob. For non-hardware-backed keys, the keyblob should be twice the
+ * length of the user key. For hardware-backed keys, the keyblob length
+ * should always be 256 bits and the caller should populate the key blob with
+ * their desired key version and salt value. The first 32 bits of the key blob
+ * are interpreted in little-endian form as the version, and the remaining
+ * 224 bits are concatenated with the one-word key mode to become the salt.
```

Comment:
I think this function should not be used to create handles for hardware-backed keys, since no key material is being imported in that case. This use case should probably call `otcrypto_hw_backed_key` function above instead.

Reply:
Agree. Updated the header comment accordingly.